src/libraries/System.Private.CoreLib/src/System/Globalization/CultureData.cs

```diff
@@ -1141,32 +1100,23 @@ internal string SpecificCultureName
                                                                             IcuGetThreeLetterWindowsLanguageName(_sRealName!);
 
         /// <summary>
-        /// Localized name for this language (Windows Only) ie: Inglis
-        /// This is only valid for Windows 8 and higher neutrals:
+        /// Localized name for this language
         /// </summary>
         private string LocalizedLanguageName
         {
             get
             {
-                if (_sLocalizedLanguage == null && !GlobalizationMode.Invariant)
+                string sLocalizedLanguage = NativeLanguageName;
+                if (!GlobalizationMode.Invariant && Name.Length > 0)
                 {
-                    // Usually the UI culture shouldn't be different than what we got from WinRT except
-                    // if DefaultThreadCurrentUICulture was set
-                    CultureInfo ci;
-
-                    if (CultureInfo.DefaultThreadCurrentUICulture != null &&
-                        ((ci = CultureInfo.GetUserDefaultCulture()) != null) &&
-                        !CultureInfo.DefaultThreadCurrentUICulture!.Name.Equals(ci.Name))
+                    // If ICU is enabled we call it anyway. If NLS, we call it only if the Windows UI language match the Current UI language
+                    if (!GlobalizationMode.UseNls || CultureInfo.UserDefaultUICulture?.Name == CultureInfo.CurrentUICulture.Name)
```

Comment:
The original check was checking `DefaultThreadCurrentUICulture`, but we are no longer checking that. Is that intentional?

Reply:
Yes, that is right. The old logic was wrong especially for Windows. We should always check `CurrentUICulture` instead.